subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -493,39 +472,90 @@ static int cmd_bap_broadcast_assistant_add_src(const struct shell *sh,
 
 static bool broadcast_source_found(struct bt_data *data, void *user_data)
 {
+	static char identified_addr_str[BT_ADDR_LE_STR_LEN];
+	static uint32_t broadcast_id = INVALID_BROADCAST_ID;
 	struct bt_bap_broadcast_assistant_add_src_param param = { 0 };
 	const struct bt_le_scan_recv_info *info = user_data;
 	char addr_str[BT_ADDR_LE_STR_LEN];
 	struct bt_uuid_16 adv_uuid;
-	uint32_t broadcast_id;
+	bool identified_broadcast = false;
 	int err;
 
 	/* Verify that it is a BAP broadcaster*/
 
-	if (data->type != BT_DATA_SVC_DATA16) {
-		return true;
-	}
+	bt_addr_le_to_str(info->addr, addr_str, sizeof(addr_str));
 
-	if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
-		return true;
-	}
+	switch (data->type) {
+	case BT_DATA_SVC_DATA16:
+		if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+			return true;
+		}
 
-	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
-		return true;
-	}
+		if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+			return true;
+		}
+
+		if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+			return true;
+		}
+
+		broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+		auto_scan.identified_id = true;
+
+		if (broadcast_id == auto_scan.broadcast_id) {
+			identified_broadcast = true;
+		}
+
+		if (auto_scan.identified_name) {
+			if (!strncmp(identified_addr_str, addr_str, strlen(addr_str))) {
+				identified_broadcast = true;
+			}
+		} else {
```

Comment:
If we are scanning for name, we should store the broadcast_id so that we can use it when `data->type == BT_DATA_BROADCAST_NAME`

Reply:
the `broadcast_id` and `identified_addr_str` are static variables, so we have them  when `data->type == BT_DATA_BROADCAST_NAME` (in case of receiving `broadcast_id` is before `broadcast_name`).

Reply:
Oh, missed that they were static. 

But that also means that the local (static) `broadcast_id` will never be assigned to `broadcast_id = INVALID_BROADCAST_ID` again after the first call. Doesn't that break things? 

Why are they static in the callback, rather than being added to `bt_auto_scan`?

Reply:
No, it can be updated if we have a new valid broadcast_id, but just applied when we found the `broadcast_name` and `broadcast_id` in the same address, or find `(broadcast_id == auto_scan.broadcast_id)`.
About adding the broadcast_id to the bt_auto_scan, I remember it was your suggestion (and also Jens's according to his comment) to keep the global parameters as minimum as possible. But anyway, I don't think there is any difference to define it globally or locally static in functionality (but maybe would be a little better in clarity).

Reply:
> But anyway, I don't think there is any difference to define it globally or locally static in functionality (but maybe would be a little better in clarity).

The difference is that it's more clearly related to the auto_scan, as well as being properly cleared on scan timeout.

Reply:
I agree, it was also my first suggestion...
Furthermore, besides the `broadcast_id`, I also prefer to add the `identified_addr_str` to the `auto_scan`. Do you agree?

Reply:
Yeah, I think that makes sense

Reply:
Alternatively you do as I suggested with how the BAP broadcast assistant sample does it, but just have a local (non-static) variable in the receive callback before the call to `bt_data_parse` if you want save some static memory at the cost of some stack memory